src/golang/cmd/server/handler/edit_dynamic_engine.go

```diff
@@ -79,7 +82,7 @@ func (*EditDynamicEngineHandler) Prepare(r *http.Request) (interface{}, int, err
 
 	return &editDynamicEngineArgs{
 		AqContext:     aqContext,
-		action:        action,
+		action:        dynamicEngineAction(action),
```

Comment:
Can you check that the action string is a valid right before this? We can make a simple helper above called `bool isValidDynamicAction(action string)`.